### PR TITLE
Add Hyperparameters Selection

### PR DESCRIPTION
**Pull request by samborba**
- Add new cell description for hyperparameters selection
- Add dict of essential parameters among those offered by the class of each model
- Change the class constructor call by passing the dict destruct with the selected hyperparameters as an argument

Task_ID: PLATIA-629

### CODE REVIEW

File: samples/mlp-classifier/Training.ipynb [hunk 1]
```diff
@@ -42,6 +42,39 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"hidden_layer_sizes\": (100, ), #@param {type:\"tuple\"}\n",
+    "    \"activation\": \"relu\", #@param {type:\"string\"}\n",
+    "    \"solver\": \"adam\", #@param {type:\"string\"}\n",
+    "    \"learning_rate\": \"constant\", #@param {type:\"string\"}\n",
+    "    \"max_iter\": 200, #@param {type:\"int\"}\n",
+    "    \"shuffle\": True, #@param {type: \"bool\"}\n",
```
Comment (fberanizo) [inline samples/mlp-classifier/Training.ipynb:74]: Adicione label e description ao hidden_layer_sizes, activation, solver, learning_rate, max_iter, shuffle (em português).
Adicione lista de opções para:
```
activation{‘identity’, ‘logistic’, ‘tanh’, ‘relu’}, default=’relu’
solver{‘lbfgs’, ‘sgd’, ‘adam’}, default=’adam’
learning_rate{‘constant’, ‘invscaling’, ‘adaptive’}, default=’constant’
```

File: samples/mlp-regressor/Training.ipynb [hunk 2]
```diff
@@ -43,6 +43,39 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"hidden_layer_sizes\": (100, ), #@param {type:\"tuple\"}\n",
+    "    \"activation\": \"relu\", #@param {type:\"string\"}\n",
+    "    \"solver\": \"adam\", #@param {type:\"string\"}\n",
+    "    \"learning_rate\": \"constant\", #@param {type:\"string\"}\n",
+    "    \"max_iter\": 200, #@param {type:\"int\"}\n",
+    "    \"shuffle\": True, #@param {type: \"bool\"}\n",
```
Comment (fberanizo) [inline samples/mlp-regressor/Training.ipynb:75]: Adicione label e description ao hidden_layer_sizes, activation, solver, learning_rate, max_iter, shuffle (em português).

Adicione lista de opções para:
```
activation{‘identity’, ‘logistic’, ‘tanh’, ‘relu’}, default=’relu’
solver{‘lbfgs’, ‘sgd’, ‘adam’}, default=’adam’
learning_rate{‘constant’, ‘invscaling’, ‘adaptive’}, default=’constant’
```

File: samples/normalizer/Training.ipynb [hunk 3]
```diff
@@ -41,6 +41,32 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"norm\": \"l2\", #@param {type:\"string\"}\n",
```
Comment (fberanizo) [inline samples/normalizer/Training.ipynb:67]: Adicione label e description ao norm (em português).

Adicione lista de opções para:
```
norm‘l1’, ‘l2’, or ‘max’, optional (‘l2’ by default)
```

File: samples/random-forest-classifier/Training.ipynb [hunk 4]
```diff
@@ -43,6 +43,37 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"n_estimators\": 100, #@param {type:\"int\"}\n",
+    "    \"criterion\": \"gini\", #@param {type:\"string\"}\n",
+    "    \"max_depth\": None, #@param {type:\"int\"}\n",
+    "    \"max_features\": \"auto\", #@param {type:[\"string\", \"int\", \"float\"]}\n",
+    "    \"class_weight\": None, #@param {type:[\"string\", \"dict\", \"list\"]}\n",
```
Comment (fberanizo) [inline samples/random-forest-classifier/Training.ipynb:73]: Adicione label e description ao n_estimators, criterion, max_depth, max_features, class_weight (em português).

Adicione lista de opções para:
```
criterion{“gini”, “entropy”}, default=”gini”
```

Mude class weight para  `#@param ["balanced”, “balanced_subsample”] {type:"string"}`
pois a interface ainda não permite passar valores de vários tipos (string, dict, list)

File: samples/random-forest-regressor/Training.ipynb [hunk 5]
```diff
@@ -39,8 +39,37 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"n_estimators\": 10, #@param {type:\"int\"}\n",
+    "    \"criterion\": \"mse\", #@param {type:\"string\"}\n",
+    "    \"max_depth\": None, #@param {type:\"int\"}\n",
+    "    \"max_features\": \"auto\", #@param {type: [\"string\", \"int\", \"float\"]}\n",
+    "}"
```
Comment (fberanizo) [inline samples/random-forest-regressor/Training.ipynb:72]: Adicione label e description ao n_estimators, criterion, max_depth, max_features  (em português).

Adicione lista de opções para:
```
criterion{“gini”, “entropy”}, default=”gini”
```

File: samples/random-forest-regressor/Training.ipynb [hunk 6]
```diff
@@ -320,6 +349,15 @@
...
     "save_model(pipeline=pipeline, columns=columns)"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline.predict([[-122.22, 37.86, 21.0, 7099.0, 1106.0, 2401.0, 1138.0, 8.3014, \"NEAR BAY\"]])"
```
Comment (fberanizo) [inline samples/random-forest-regressor/Training.ipynb:359]: Remova essa célula com código de teste.

File: samples/svc/Training.ipynb [hunk 7]
```diff
@@ -43,6 +43,39 @@
...
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"C\": 1.0, #@param {type:\"float\"}\n",
+    "    \"kernel\": \"rbf\", #@param {type:\"string\"}\n",
+    "    \"degree\": 3, #@param {type:\"int\"}\n",
+    "    \"gamma\": \"auto\", #@param {type: \"string\"}\n",
+    "    \"probability\": False, #@param {type: \"bool\"}\n",
+    "    \"max_iter\": -1, #@param {type:\"int\"}\n",
+    "}"
```
Comment (fberanizo) [inline samples/svc/Training.ipynb:76]: Adicione label e description ao C, kernel, degree, gamma, probability, max_iter (em português).

Adicione lista de opções para:
```
kernel{‘linear’, ‘poly’, ‘rbf’, ‘sigmoid’, ‘precomputed’}, default=’rbf’
gamma{‘scale’, ‘auto’}, default=’scale’
```

File: samples/svc/Training.ipynb [hunk 8]
```diff
@@ -165,12 +198,30 @@
...
+    "from sklearn.model_selection import GridSearchCV\n",
+    "\n",
+    "grid = GridSearchCV(estimator=pipeline.named_steps.estimator,\n",
+    "                    param_grid={'gamma': [\"auto\"],\n",
+    "                                'probability': [True]},\n",
+    "                    cv=5)\n",
+    "\n",
+    "grid.fit(X_train, y_train)\n",
+    "\n",
+    "grid.cv_results_"
```
Comment (fberanizo) [inline samples/svc/Training.ipynb:222]: O objetivo aqui era fazer Grid Search ou CrossValidation?
Parece que não faz busca alguma, já que só tem 1 combinação de parâmetros possível....

File: samples/svr/Training.ipynb [hunk 9]
```diff
@@ -43,6 +43,37 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameters = {\n",
+    "    \"C\": 1.0, #@param {type:\"float\"}\n",
+    "    \"kernel\": \"rbf\", #@param {type:\"string\"}\n",
+    "    \"degree\": 3, #@param {type:\"int\"}\n",
+    "    \"gamma\": \"auto\", #@param {type: \"string\"}\n",
+    "    \"max_iter\": -1, #@param {type:\"int\"}\n",
```
Comment (fberanizo) [inline samples/svr/Training.ipynb:73]: Adicione label e description ao C, kernel, degree, gamma, max_iter (em português).

Adicione lista de opções para:
```
kernel{‘linear’, ‘poly’, ‘rbf’, ‘sigmoid’, ‘precomputed’}, default=’rbf’
gamma{‘scale’, ‘auto’}, default=’scale’
```

File: samples/svc/Training.ipynb (hunk 8 above)
Comment (samborba) [reply samples/svc/Training.ipynb:222]: Eu estava testando algo e acabei esquecendo de remover

File: samples/logistic-regression/Training.ipynb [hunk 10]
```diff
@@ -72,15 +50,18 @@
...
-    "}"
+    "# parameters\n",
+    "dataset = \"iris\" #@param {type:\"string\"}\n",
+    "target = \"Species\" #@param {type:\"string\"}\n",
+    "\n",
+    "# hyperparameters\n",
+    "penalty = \"l2\" #@param [\"l1, \"l2\", \"elasticnet\", \"None\"] {type:\"string\", label:\"Penalidade\", description:\"Norma utilizada na penalização do erro\"}\n",
+    "C = 1.0 #@param {type:\"number\", label:\"Regularização Inversa\", description:\"Retém a modificação de força da regularização ao ser posicionada inversamente no regulador Lambda\"}\n",
+    "fit_intercept = True #@param {type\"boolean\", label:\"Interceptação\", description:\"Especifica se uma constante (viés ou interceptação) deve ser adicionada à função de decisão\"}\n",
+    "class_weight = None #@param [\"balanced”, “balanced_subsample”] {type:\"string\", label:\"Peso das Classes\", description:\"Especifica pesos de amostras quando for ajustar classificadores como uma função da classe do target\"}\n",
```
Comment (fberanizo) [inline samples/logistic-regression/Training.ipynb:61]: Parece que aqui está com aquelas aspas curvas “ ” tanto que no código fonte não está \\"

File: samples/random-forest-classifier/Training.ipynb [hunk 11]
```diff
@@ -39,8 +48,16 @@
...
     "dataset = \"iris\" #@param {type:\"string\"}\n",
-    "target = \"Species\" #@param {type:\"string\"}"
+    "target = \"Species\" #@param {type:\"string\"}\n",
+    "\n",
+    "# hyperparameters\n",
+    "n_estimators = 10 #@param {type:\"integer\", label:\"Estimadores\", description:\"Número de árvores na floresta\"}\n",
+    "criterion = \"gini\" #@param [\"gini\", \"entropy\"] {type:\"string\", label:\"Critério\", description:\"Função para medir a qualidade de uma divisão\"}\n",
+    "max_depth = None #@param {type:\"integer\", label:\"Profundidade\", description:\"O máximo de profundidade da árvore\"}\n",
+    "max_features = \"auto\" #@param [\"auto\", \"sqrt\", \"log2\"] {type:\"string\", label:\"Features\", description:\"O máximo de features a serem considerados ao procurar a melhor divisão\"}\n",
+    "class_weight = None #@param [\"balanced”, “balanced_subsample”] {type:\"string\", label:\"Peso das Classes\", description:\"Especifica pesos de amostras quando for ajustar classificadores como uma função da classe do target\"}"
```
Comment (fberanizo) [inline samples/random-forest-classifier/Training.ipynb:60]: Parece que aqui está com aquelas aspas curvas “ ” tanto que no código fonte não está \\"
